### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by biphobe**
I followed the guide on Windows so I thought I would update the `CONTRIBUTING.md` a bit. 

I tried to make readable, small commits, but I'm just adding my reasoning since commit names length is limited:
- document mentioned 2 processes to use (copy on file change/use local registry) for working with Redwood, but only the `copy:watch` one included informations about actual building process of Redwood,
  - so I've reworked on the whole section to make clear distinction for build process in the beginning (since it's same for both "approaches") to make a separate instructions afterwards,
- `rwt` was mentioned in multiple sections duplicating similar informations,
  - so I aggregated all of these and moved them to already existing section for `rwt`
- the `please read and understand our CoC` had a passive aggresive vibe to it,
  - so I rephrase it to `please check our CoC first`
- Windows-related informations were needlessly worrisome since apart from the https://github.com/redwoodjs/redwood/issues/701 I consider setting up the Redwood's monorepo on Windows a pleasant experience,
  - I've added information about upgrading to WSL 2 since WSL 1 didn't support `lsof` and would cause issues with, e.g. [tasks/publish-local#L11](https://github.com/redwoodjs/redwood/blob/main/tasks/publish-local#L11),
  - I've rephrased overly expressive information about Windows' `copy:watch` issue (linked issue with a short description is enough and it's contents shouldn't be copied onto the contributions docs)
  - I've removed the "you may need rsync" information as most popular distros, including those supported by WSL, already include rsync by default,
- Information for setting `RW_PATH` env variable was weirdly duplicated onto separate Linux and Mac sections even though it were basically same informations,
  - so I unified this and removed request for adding similar entry for Windows since setting these while using WSL is the same process as in any linux (even though I would argue if section for explaining how to add env variables is needed at all)
- `rwt` command was listed on the same level of nesting as `rwt copy` implicating that `rwt` a command of the same category/group as `rwt copy` which is untrue,
  - so I fixed the nested headers to make explicit distinction between `rwt` in general and it's commands

And: fixed a a few typos, errors (e.g. untrue information that `publish-local` builds the app), inconsistencies (upper/lower cases, phrasings, quotes etc) and needlessly repeated terminal outputs for example or two.

### CODE REVIEW

**Comment by Tobbe** [comment]
> I followed the guide on Windows so I thought I would update the CONTRIBUTING.md a bit.

Thank you for taking the time to do all of this! New-user insights are super valuable :)

Since I'm a Windows user myself, I paid extra attention to updates regarding Windows support. With all the updates/comments you've made, I'm starting to wonder if we should have separate comments for WSL and Git-for-Windows setups. What do you think?

> I've removed the "you may need rsync" information as most popular distros, including those supported by WSL, already include rsync by default,

I'd prefer we keep this. It's very much needed for everyone setting up Redwood on Windows using Git-for-Windows and not WSL.

**Comment by peterp** [comment]
> I'm starting to wonder if we should have separate comments for WSL and Git-for-Windows setups. What do you think?

I think that would be awesome! I'm wondering if we shouldn't consider creating separate documents for Windows and *nix?

**Comment by peterp** [comment]
@thedavidprice @jtoar Could I ask either of you to weight in on these changes, a bunch of it seems like decent improvements from a beginners point of view, but there's also a lot of stylistic differences, and I'm not sure if they align with what you want to achieve.

**Comment by thedavidprice** [comment]
@jtoar I've added you as the Reviewer. Available if you need me, but I'll assume you're good to review + merge unless I hear otherwise.

Thanks all!

**Comment by jtoar** [comment]
@biphobe I can't apologize enough for the delay; more than dropped the ball in getting back to you.

I agree with many of the changes you made. A beginner's point of view is invaluable, and in so far as reading is harder than writing (which it is!) I really appreciate your point of view, especially as a Windows user, which (as a contributor!) is hard enough.

I'm including some of the changes you made that I think are constructive in another PR. Why another PR and not this one? Basically, this PR is huge. It changes a lot, and until I became a maintainer, I didn't know why that mattered. So, an inability on my end rather than yours! 

Responding to you bullet-by-bullet, where I felt I had something to say (so just two things):

- the please read and understand our CoC had a passive aggresive vibe to it

Redwood has a really great community, and I would rather be too aggressive with having people read and understand (because those aren't the same thing!) than not. "Check" sounds too laissez-faire, like "did the tests pass? great, there must be no bugs!" rather than "what did we actually test for?"

- [something was] weirdly duplicated

Many people try to read as little as they can, and (given a table of contents) that means jumping straight to a certain section. So duplicating information isn't a mistake, it's often intentional. Moreover, we try to be as terse as we can with code, but that's only because we're talking to computers. When it comes to explaining something, we're talking to people instead. So I'm trying to address a few different scenarios here.

I meant none of that disparagingly. And many of the changes you suggested are right, so thank you for making contributing to Redwood better, which is a huge since that's an activity that cascades!
